doc/nvim-tree-lua.txt

```diff
@@ -951,7 +952,97 @@ You'll also need to set |nvim-tree.hijack_netrw| to `true` during setup.
 A good functionnality to enable is |nvim-tree.hijack_directories|.
 
 ==============================================================================
- 5. MAPPINGS                                              *nvim-tree-mappings*
+ 5. API                                                   *nvim-tree-api*
+
+Nvim-tree's public API can be used to access features.
+>
+    local nt_api = require("nvim-tree.api")
+<
+This module exposes stable functionalities, it is advised to use this in order
+to avoid breaking configurations due to internal breaking changes.
+
+The api is separated in multiple modules, which can be accessed with
+`require("nvim-tree.api").moduleName.functionality`.
+
+Functions that needs a tree node parameter are exposed with an abstraction
```

Comment:
This is a little unclear right now, however it will become clear after we add on_attach and its mappings.